src/mesh.cpp

```diff
@@ -1367,15 +1388,14 @@ extern "C" int openmc_mesh_get_type(int32_t index, char* type)
   if (int err = check_mesh(index))
     return err;
 
-  RegularMesh* mesh = dynamic_cast<RegularMesh*>(model::meshes[index].get());
-  if (mesh) {
+  if (is_mesh_type<RegularMesh>(index)) {
     std::strcpy(type, "regular");
-  } else {
-    RectilinearMesh* mesh =
-      dynamic_cast<RectilinearMesh*>(model::meshes[index].get());
-    if (mesh) {
-      std::strcpy(type, "rectilinear");
-    }
+  } else if (is_mesh_type<RectilinearMesh>(index)) {
```

Comment:
It is better to just add a virtual method to each mesh class that returns a descriptive string (char* in this case, not std::string) for its name rather than doing this dynamic casting.